server/src/main/java/org/elasticsearch/common/settings/Settings.java

```diff
@@ -99,7 +100,43 @@ private static Settings of(Map<String, Object> settings, SecureSettings secureSe
 
     private Settings(Map<String, Object> settings, SecureSettings secureSettings) {
         // we use a sorted map for consistent serialization when using getAsMap()
-        this.settings = Collections.unmodifiableNavigableMap(new TreeMap<>(settings));
```

Comment:
I think I would prefer this in `Builder.build()`. Seems more logical and also, AFAICS, that is the only path we need to cover.

Reply:
I found it a little easier to reason about to keep it here, so we are always guaranteed to get a fully immutable structure now (also for lists etc.).
Also technically `org.elasticsearch.common.settings.Settings#getByPrefix`
which uses `Settings.of(` profits from this change as well so maybe just keep it here in the constructor to keep things simple and have this apply in all cases? We could look into some optimizations in a follow-up maybe, but generally as discussed elsewhere setting setup isn't a hot path issue.

Reply:
I mainly found it illogical that we do not fully build the parameters for the constructor in the build method like we normally do. I would still prefer it the other way around, but can accept it residing here for now.

I am not sure I found a place where `getByPrefix` needs this, they all look like "get and throw away" style lookups, perhaps you have a specific case. Even if there are such cases, it seems unnecessary to go through the lists once more when all we need is interned keys in that case.